internal/headers.go

```diff
@@ -36,8 +36,25 @@ type HeaderReader interface {
 	ForEachKey(handler func(string, []byte) error) error
 }
 
-// ContextPropagator is an interface that determines what information from
-// context to pass along
+// ContextPropagator determines what information from context to pass along.
+// Returning an error from any of its methods will prevent the normal execution
```

Comment:
This is actually not true. At least this Inject method will fail open.

Reply:
@alshopov Could you update the doc here? Or should we handle it on our side?